storage/interceptors/secured_generate_credentials_interceptor.go

```diff
@@ -50,7 +51,11 @@ func (c *generateCredentialsInterceptor) AroundTxCreate(h storage.InterceptCreat
 			log.C(ctx).Error("Could not generate credentials for platform")
 			return nil, err
 		}
-		(obj.(types.Secured)).SetCredentials(credentials)
+		platform, ok := obj.(*types.Platform)
```

Comment:
This interceptor is assigned only to platform type, but the interceptor is generic in order to allow generation of credentials for arbitrary type.
One option is to make this interceptor only for platforms (name of the filter).
Another option is to introduce another interface to set credentials of certain types object and here check for this interface implementation. If implemented set the credentials of the object via the same interface.

Reply:
I will rename the filter so it is platform specific